### PR TITLE
Tarmac reference building

### PR DESCRIPTION
**Pull request by sndrtj**

### CODE REVIEW

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala [hunk 1]
```diff
@@ -33,7 +53,93 @@ class Tarmac(val parent: Configurable) extends QScript with PedigreeQscript with
   }
 
   def addMultiSampleJobs() = {
+    val initRefMap = samples map { x => x._2 -> getReferenceSamplesForSample(x._1) }
```
Comment (bow) [inline tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala]: to make this more readable, you can also pattern match on the tuple directly:
```
val initRefMap = samples.map { case (a, b) => b -> getReferenceSamplesForSample(a) }
```
(of course `a` and `b` should be named appropriately).

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala [hunk 2]
```diff
@@ -33,7 +53,93 @@ class Tarmac(val parent: Configurable) extends QScript with PedigreeQscript with
...
+    initRefMap.values.filter(_.isLeft).foreach {
+      case -\/(message) => Logging.addError(message)
+      case _            => ;
+    }
```
Comment (bow) [inline tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala]: You can shorten this like so:
```
initRefMap.values.collect { case -\/(err) => err }.foreach { Logging.addError }
```

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala [hunk 3]
```diff
@@ -33,7 +53,93 @@ class Tarmac(val parent: Configurable) extends QScript with PedigreeQscript with
...
+
+    val refMap: Map[Sample, Set[Sample]] = initRefMap.filter(_._2.isRight).map { x =>
```
Comment (bow) [inline tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala]: Similar to the above, I think using `.collect` and pattern matching would make this more clear.

**Review by bow — CHANGES_REQUESTED** [review]
Hey @sndrtj,

Looks ok in general, but I see several places that `\/` is filtered then on the subsequent map it is extracted manually. This can be combined to a single operation using `.collect`.

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala [hunk 4]
```diff
@@ -33,7 +53,93 @@ class Tarmac(val parent: Configurable) extends QScript with PedigreeQscript with
   }
 
   def addMultiSampleJobs() = {
+    val initRefMap = samples map { x => x._2 -> getReferenceSamplesForSample(x._1) }
+    initRefMap.values.filter(_.isLeft).foreach {
+      case -\/(message) => Logging.addError(message)
+      case _            => ;
```
Comment (ffinfo) [inline tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala]: `;` is redundant

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala (hunk 3 above)
Comment (ffinfo) [reply tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala]: `.map` works here but maybe do is like this:
``` scala
.map { case (key, value) => 
  // your code
}
```

**Review by ffinfo — CHANGES_REQUESTED** [review]

**Comment by ffinfo** [comment]
I think all test should be unit tested, is there a way to do this already on this branch?

File: tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala (hunk 4 above)
Comment (sndrtj) [reply tarmac/src/main/scala/nl/lumc/sasc/biopet/pipelines/tarmac/Tarmac.scala]: Compiler gives a warning message if not.

**Review by ffinfo — APPROVED** [review]

**Pull request merged**